info/features.go

```diff
@@ -0,0 +1,22 @@
+package info
+
+type Component struct {
+	Name     string
+	Features []string
+}
+
+// ServerComponent describes server component info, like implemented features.
+var ServerComponent = Component{
+	Name: "server",
+	Features: []string{
+		"features",
```

Comment:
how and when is this going to be populated?

Reply:
They are going to be populated manually for now. 

When a new feature that the control plane needs to be aware of is developed, it will be added here.

Following your comment here https://github.com/rudderlabs/rudder-server/pull/2377#discussion_r964753658, I found this approach to be the most straightforward and cleanest for now.

The dynamic enabled/disable of features based on config is not supported yet, but it can be easily implemented if required by something like this:

```go
func ServerComponentFromConfig(conf config.Config) Component {

  comp := Component{
	Name: "server",
	Features: []string{},
  }
  if config.IsSet("feature.enabled") {
  	comp.Features = append(comp.Features, "features")
  }
  
  return comp
}
```